src/gateway_gatt_service.erl

```diff
@@ -62,12 +64,22 @@ handle_info({changed_wifi_pass, Value}, State=#state{}) ->
                           connman:disconnect(wifi, Name)
                   end, OnlineWifiPaths),
     lager:info("Trying to connect to WiFI SSID: ~p", [State#state.ssid]),
-    case connman:connect(wifi, State#state.ssid, binary_to_list(Value), self()) of
-        ok -> ok;
-        Other ->
-            lager:notice("Start connect for SSID ~p: ~p", [State#state.ssid, Other])
-    end,
-    {noreply, State};
+    %% Start the connman agent if it was not already started.On
+    %% failure to start the agent we try to restart it later and
+    %% re-attempt the connect.
+    case connman:start_agent() of
+        ok ->
+            case connman:connect(wifi, State#state.ssid, binary_to_list(Value), self()) of
+                ok -> ok;
+                Other ->
+                    lager:notice("Start connect for SSID ~p: ~p", [State#state.ssid, Other])
+            end,
+            {noreply, State};
+        {error, Error} ->
+            lager:warning("Failed to start connman agent; ~p", [Error]),
+            erlang:send_after(?CONNMAN_AGENT_RETRY, self(), Msg),
```

Comment:
The retries should keep gateway_config from crashing but I'm guessing the hotspot will never connect to a WiFi network if the agent can't start.  Are the agent start failures a bug in connman itself?

Reply:
I think starting the agent can take a long time if connman is still busy dealing with the barrage of WiFi network info. We just don't handle it taking a while. 

Moving it here makes it so that it's likely that connman will have quieted down by this code. 

If the agent can't start the user really can't do anything anyway?